### DISCUSSION TITLE
Invalid async_mode specified

### DISCUSSION THREAD

**Discussion by daerlang** [miguelgrinberg/Flask-SocketIO#2123] [Q&A]
i faced the issue "Invalid async_mode specified". need some help.
run my app in dev env is ok but pyinstaller to exe to run is error as below:

Traceback (most recent call last): File "app\vncview.py", line 578, in <module> File "flask_socketio\__init__.py", line 187, in __init__ File "flask_socketio\__init__.py", line 243, in init_app File "socketio\base_server.py", line 31, in __init__ File "engineio\base_server.py", line 81, in __init__ ValueError: Invalid async_mode specified   

my pip list as bellow:
`(vncview_env) PS D:\python> pip list
Package Version

aiohappyeyeballs 2.4.4
aiohttp 3.11.9
aiosignal 1.3.1
altgraph 0.17.4
async-timeout 4.0.3
attrs 24.2.0
bidict 0.23.1
blinker 1.9.0
certifi 2024.8.30
cffi 1.17.1
charset-normalizer 3.4.0
click 8.1.7
colorama 0.4.6
cryptography 44.0.0
dnspython 2.7.0
eventlet 0.38.1
Flask 3.1.0
Flask-SocketIO 5.4.1
frozenlist 1.5.0
gevent 24.11.1
gevent-websocket 0.10.1
greenlet 3.1.1
h11 0.14.0
idna 3.10
itsdangerous 2.2.0
Jinja2 3.1.4
JsonForm 0.0.2
jsonschema 4.23.0
jsonschema-specifications 2024.10.1
JsonSir 0.0.2
jwcrypto 1.5.6
MarkupSafe 3.0.2
multidict 6.1.0
mysql-connector-python 9.1.0
numpy 2.1.3
nvidia-ml-py 12.560.30
packaging 24.2
pefile 2023.2.7
pillow 11.0.0
pip 24.3.1
propcache 0.2.1
psutil 6.1.0
py3nvml 0.2.7
pycparser 2.22
pyinstaller 6.11.1
pyinstaller-hooks-contrib 2024.10
pyngrok 7.2.1
pynvml 12.0.0
Python-EasyConfig 0.1.7
python-engineio 4.10.1
python-socketio 5.11.4
pywin32 308
pywin32-ctypes 0.2.3
PyYAML 6.0.2
redis 5.2.0
referencing 0.35.1
requests 2.32.3
Resource 0.2.1
rpds-py 0.22.1
setuptools 65.5.0
simple-websocket 1.1.0
six 1.16.0
urllib3 2.2.3
wakeonlan 3.1.0
Werkzeug 3.1.3
WMI 1.5.1
wsproto 1.2.0
xmltodict 0.14.2
yarl 1.18.3
zope.event 5.0
zope.interface 7.2`

pyinstaller include

25275 DEBUG: Hidden import 'engineio' already found
25275 DEBUG: Hidden import 'simple_websocket' already found`
my flask code:
app = Flask(__name__, static_folder=r'D:\python\vncview\app\noVNC-1.5.0', static_url_path='/') socketio = SocketIO(app)

**Comment by miguelgrinberg** [comment]
Have you read through https://github.com/miguelgrinberg/python-socketio/issues/35? That's a long discussion among pyinstaller users. Unfortunately I do not use pyinstaller myself. The linked issue collects all the resources available on this topic.
